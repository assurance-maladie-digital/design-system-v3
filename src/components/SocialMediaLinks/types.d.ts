export interface SocialMediaLink {
	icon: string
	name: string
	href: string
}
