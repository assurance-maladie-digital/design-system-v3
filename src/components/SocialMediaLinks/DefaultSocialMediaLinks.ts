import { mdiTwitter, mdiLinkedin } from '@mdi/js'

import type { SocialMediaLink } from './types'

export const defaultSocialMediaLinks: SocialMediaLink[] = [
	{
		icon: mdiLinkedin,
		href: 'https://www.linkedin.com/company/assurance-maladie/',
	},
	{
		icon: mdiTwitter,
		href: 'https://twitter.com/Assur_Maladie',
	},
]
