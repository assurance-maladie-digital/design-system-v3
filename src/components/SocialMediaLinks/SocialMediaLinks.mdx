import {Controls, Canvas, Meta, Source} from '@storybook/blocks';

import * as SocialMediaLinksStories from './SocialMediaLinks.stories.ts';

<Meta of={SocialMediaLinksStories} />

Le composant `SocialMediaLinks` est utilis√© pour afficher des liens de reseaux sociaux dans le componant `Footer`.


<Canvas of={SocialMediaLinksStories.Default}  />

# API

<Controls of={SocialMediaLinksStories.Default} />