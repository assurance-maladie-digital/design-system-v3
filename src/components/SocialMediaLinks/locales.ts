export const locales = {
	followUs: 'Suivez-nous',
}
