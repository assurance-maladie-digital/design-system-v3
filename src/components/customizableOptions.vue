// src/components/customizableOptions.vue
<script setup>
	import useCustomizableOptions from '../composables/useCustomizableOptions'

	const defaultOptions = {
		btn: {
			color: 'primary',
		},
	}

	const props = defineProps({
		vuetifyOptions: {
			type: Object,
			default: () => ({}),
		},
	})

	const options = useCustomizableOptions(defaultOptions, props)
</script>

<template>
	<div>
		<VBtn v-bind="options.btn">
			Button
		</VBtn>
	</div>
</template>
