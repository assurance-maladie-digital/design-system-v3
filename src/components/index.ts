export { default as Alert } from './Alert/Alert.vue'
export { default as BackBtn } from './BackBtn/BackBtn.vue'
export { default as BackToTopBtn } from './BackToTopBtn/BackToTopBtn.vue'
export { default as CollapsibleList } from './CollapsibleList/CollapsibleList.vue'
export { default as CopyBtn } from './CopyBtn/CopyBtn.vue'
export { default as CustomInputSelect } from './Customs/CustomInputSelect/CustomInputSelect.vue'
export { default as CustomSelect } from './Customs/CustomSelect/CustomSelect.vue'
export { default as CustomTextField } from './Customs/CustomTextField/CustomTextField.vue'
export { default as DataList } from './DataList/DataList.vue'
export { default as DataListGroup } from './DataListGroup/DataListGroup.vue'
export { default as DownloadBtn } from './DownloadBtn/DownloadBtn.vue'
export { default as FooterBar } from './FooterBar/FooterBar.vue'
export { default as FranceConnectBtn } from './FranceConnectBtn/FranceConnectBtn.vue'
export { default as HeaderBar } from './HeaderBar/HeaderBar.vue'
export { default as HeaderBurgerMenu } from './HeaderBar/HeaderBurgerMenu/HeaderBurgerMenu.vue'
export { default as HeaderMenuItem } from './HeaderBar/HeaderBurgerMenu/HeaderMenuItem/HeaderMenuItem.vue'
export { default as HeaderMenuSection } from './HeaderBar/HeaderBurgerMenu/HeaderMenuSection/HeaderMenuSection.vue'
export { default as HeaderSubMenu } from './HeaderBar/HeaderBurgerMenu/HeaderSubMenu/HeaderSubMenu.vue'
export { default as HeaderLogo } from './HeaderBar/HeaderLogo/HeaderLogo.vue'
export { default as HeaderMenuBtn } from './HeaderBar/HeaderMenuBtn/HeaderMenuBtn.vue'
export { default as HeaderLoading } from './HeaderLoading/HeaderLoading.vue'
export { default as HeaderNavigationBar } from './HeaderNavigationBar/HeaderNavigationBar.vue'
export { default as LangBtn } from './LangBtn/LangBtn.vue'
export { default as Logo } from './Logo/Logo.vue'
export { default as LogoBrandSection } from './LogoBrandSection/LogoBrandSection.vue'
export { default as NotificationBar } from './NotificationBar/NotificationBar.vue'
export * from './NotificationBar/types'
export { default as PageContainer } from './PageContainer/PageContainer.vue'
export { default as SkipLink } from './SkipLink/SkipLink.vue'
export { default as SocialMediaLinks } from './SocialMediaLinks/SocialMediaLinks.vue'
export { default as SubHeader } from './SubHeader/SubHeader.vue'
