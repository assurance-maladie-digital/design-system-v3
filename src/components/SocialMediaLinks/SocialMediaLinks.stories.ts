import SocialMediaLinks from './SocialMediaLinks.vue'
import type { Meta, StoryObj } from '@storybook/vue3'
import { mdiFacebook, mdiLinkedin, mdiTwitter } from '@mdi/js'

const meta = {
	title: 'Composants/Navigation/SocialMediaLinks',
	component: SocialMediaLinks,
	parameters: {
		layout: 'fullscreen',
		controls: { exclude: ['socialMediaLinks'] },
	},
	args: {
		socialMediaLinks: [
			{
				icon: mdiLinkedin,
				href: 'https://www.linkedin.com/company/assurance-maladie/',
			},
			{
				icon: mdiFacebook,
				href: 'https://twitter.com/Assur_Maladie',
			},
			{
				icon: mdiTwitter,
				href: 'https://twitter.com/Assur_Maladie',
			},
		],
	},
	argTypes: {
		socialMediaLinks: {
			control: { type: 'object' },
		},
	},
} as Meta<typeof SocialMediaLinks>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
	parameters: {
		sourceCode: [
			{
				name: 'Template',
				code: `<template>
	<SocialMediaLinks :links=":links" />
</template>
				`,
			},
			{
				name: 'Script',
				code: `<script setup lang="ts">
	import SocialMediaLinks from '@cnamts/synapse'
	
	const links = [
		{
			icon: mdiLinkedin,
			name: 'LinkedIn',
			href: 'https://www.linkedin.com/company/assurance-maladie/',
		},
		{
			icon: mdiFacebook,
			name: 'Facebook',
			href: 'https://twitter.com/Assur_Maladie',
		},
		{
			icon: mdiTwitter,
			name: 'Twitter',
			href: 'https://twitter.com/Assur_Maladie',
		},
	]
</script>
				`,
			},
		],
	},
	args: {
		links: [
			{
				icon: mdiLinkedin,
				name: 'LinkedIn',
				href: 'https://www.linkedin.com/company/assurance-maladie/',
			},
			{
				icon: mdiFacebook,
				name: 'Facebook',
				href: 'https://twitter.com/Assur_Maladie',
			},
			{
				icon: mdiTwitter,
				name: 'Twitter',
				href: 'https://twitter.com/Assur_Maladie',
			},
		],
	},
	render: (args) => {
		return {
			components: { SocialMediaLinks },
			setup() {
				return { args }
			},
			template: `
				<div class="d-flex flex-wrap align-center pa-4">
                	<SocialMediaLinks :links="args.links" />
				</div>
            `,
		}
	},
}
