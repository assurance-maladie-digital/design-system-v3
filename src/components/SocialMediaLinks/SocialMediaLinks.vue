<script setup lang="ts">
	import type { PropType } from 'vue'
	import { locales } from './locales'
	import type { SocialMediaLink } from './types'

	const props = defineProps({
		links: {
			type: Array as PropType<SocialMediaLink[]>,
			default: null,
		},
	})
</script>

<template>
	<div class="d-flex flex-column">
		<span class="vd-social-media-links-label text-subtitle-2 text--primary">
			{{ locales.followUs }}
		</span>

		<ul class="d-flex max-width-none">
			<li
				v-for="(social, index) in props.links"
				:key="index"
			>
				<VBtn
					id="social-btn"
					:href="social.href"
					target="_blank"
					rel="noopener noreferrer"
					:icon="true"
					variant="text"
				>
					<VIcon
						size="30px"
						class="vd-social-media-links-icon"
					>
						{{ social.icon }}
					</VIcon>
				</VBtn>
			</li>
		</ul>
	</div>
</template>

<style lang="scss" scoped>
@import '../../assets/tokens.scss';

.vd-social-media-links {
  display: flex;
  flex-direction: column;
}

li {
  list-style: none;
}

.vd-social-media-links-label.text--primary {
  color: $blue-base;
}

.v-theme--dark .vd-social-media-links-label.text--primary {
  color: white;
}

.vd-social-media-links-icon {
  color: $grey-base;
}

.v-btn--icon {
  width: calc(var(--v-btn-height) + 5px);
  height: calc(var(--v-btn-height) + 5px);
  border: 0;
}

.v-theme--dark .v-btn--variant-text:hover :deep() {
  background: rgba(white, 0.1);
}

@media (min-width: 768px) {
  .vd-social-media-links-label {
    text-align: right;
  }
}

@media (max-width: 767px) {
  .vd-social-media-links-label {
    text-align: left;
  }
}
</style>
