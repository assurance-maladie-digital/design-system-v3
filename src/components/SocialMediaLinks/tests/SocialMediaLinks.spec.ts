import { mount, VueWrapper } from '@vue/test-utils'
import SocialMediaLinks from '../SocialMediaLinks.vue'
import { describe, it, expect, afterEach } from 'vitest'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

const vuetify = createVuetify({
	components,
	directives,
})

describe('SocialMediaLinks', () => {
	let wrapper: VueWrapper

	afterEach(() => {
		if (wrapper) {
			wrapper.unmount()
		}
	})

	it('renders correctly with default props', () => {
		wrapper = mount(SocialMediaLinks, {
			global: {
				plugins: [vuetify],
			},
			props: {
				links: undefined,
			},
		})
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('renders correctly with provided links', () => {
		const links = [
			{ href: 'https://twitter.com', name: 'Twitter', icon: 'mdi-twitter' },
			{ href: 'https://facebook.com', name: 'Facebook', icon: 'mdi-facebook' },
		]
		wrapper = mount(SocialMediaLinks, {
			global: {
				plugins: [vuetify],
			},
			props: {
				links,
			},
		})
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('renders the correct number of social media links', () => {
		const links = [
			{ href: 'https://twitter.com', name: 'Twitter', icon: 'mdi-twitter' },
			{ href: 'https://facebook.com', name: 'Facebook', icon: 'mdi-facebook' },
		]
		wrapper = mount(SocialMediaLinks, {
			global: {
				plugins: [vuetify],
			},
			props: {
				links,
			},
		})
		expect(wrapper.findAll('li').length).toBe(links.length)
	})

	it('renders no links when links prop is empty array', () => {
		wrapper = mount(SocialMediaLinks, {
			global: {
				plugins: [vuetify],
			},
			props: {
				links: [],
			},
		})
		expect(wrapper.findAll('li').length).toBe(0)
	})

	it('renders no links when links prop is null', () => {
		wrapper = mount(SocialMediaLinks, {
			global: {
				plugins: [vuetify],
			},
			props: {
				links: undefined,
			},
		})
		expect(wrapper.findAll('li').length).toBe(0)
	})
})
