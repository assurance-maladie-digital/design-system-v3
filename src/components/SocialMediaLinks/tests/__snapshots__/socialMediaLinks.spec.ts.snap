// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialMediaLinks.vue > renders correctly with default props 1`] = `
"<div data-v-9d3df26f="" class="d-flex flex-column"><span data-v-9d3df26f="" class="vd-social-media-links-label text-subtitle-2 text--primary">Suivez-nous</span>
  <ul data-v-9d3df26f="" class="d-flex max-width-none"></ul>
</div>"
`;

exports[`SocialMediaLinks.vue > renders correctly with provided links 1`] = `
"<div data-v-9d3df26f="" class="d-flex flex-column"><span data-v-9d3df26f="" class="vd-social-media-links-label text-subtitle-2 text--primary">Suivez-nous</span>
  <ul data-v-9d3df26f="" class="d-flex max-width-none">
    <li data-v-9d3df26f="">
      <vbtn data-v-9d3df26f="" id="social-btn" href="https://twitter.com" target="_blank" rel="noopener noreferrer" icon="true" variant="text">
        <vicon data-v-9d3df26f="" size="30px" class="vd-social-media-links-icon">mdi-twitter</vicon>
      </vbtn>
    </li>
    <li data-v-9d3df26f="">
      <vbtn data-v-9d3df26f="" id="social-btn" href="https://facebook.com" target="_blank" rel="noopener noreferrer" icon="true" variant="text">
        <vicon data-v-9d3df26f="" size="30px" class="vd-social-media-links-icon">mdi-facebook</vicon>
      </vbtn>
    </li>
  </ul>
</div>"
`;
